.github/workflows/generate-zip.yml

```diff
@@ -0,0 +1,40 @@
+name: Generate ZIP file
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+    generate-zip-file:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+
+            - name: Cache node_modules
+              id: cache-node-modules
+              uses: actions/cache@v4
+              env:
+                  cache-name: cache-node-modules
+              with:
+                  path: node_modules
+                  key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+
+            - name: Setup node version and npm cache
+              uses: actions/setup-node@v4
+              with:
+                  node-version-file: '.nvmrc'
+                  cache: 'npm'
+
+            - name: Install Node dependencies
+              if: steps.cache-node-modules.outputs.cache-hit != 'true'
+              run: npm ci --no-optional
```

Comment:
Looking at the [`build` script](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/develop/package.json#L113), it will run a `prebuild` step that will run `npm install`. Because of that, I think we can remove this line, otherwise we'll run install twice.